packages/jaeger-ui/src/components/TracePage/TraceStatistics/index.tsx

```diff
@@ -392,79 +188,159 @@ export default class TraceStatistics extends Component<Props, State> {
+      return {
+        style: { background: backgroundColor, borderColor: backgroundColor },
+      };
+    };
 
-  render() {
+    let columns: ColumnProps<ITableSpan>[] = [
+      {
+        title: 'Name',
+        dataIndex: 'name',
+        sorter: sorterFunction('name'),
+        render: (name: string, row: ITableSpan) => {
+          return (
+            <span
+              onClick={() => onClickOption(row.type, row.name)}
+              style={{
+                borderLeft: `4px solid ${`${row.color}` || `transparent`}`,
+                padding: '7px 0px 7px 10px',
+                cursor: 'default',
+              }}
+            >
+              {name}
+            </span>
+          );
+        },
+      },
+      {
+        title: 'Count',
+        dataIndex: 'count',
+        sorter: sorterFunction('count'),
+        defaultSortOrder: 'ascend',
+      },
+      {
+        title: 'Total',
+        dataIndex: 'total',
+        sorter: sorterFunction('total'),
+        render: (cell: string) => {
+          return `${cell} ms`;
+        },
+      },
+      {
+        title: 'Avg',
+        dataIndex: 'avg',
+        sorter: sorterFunction('avg'),
+        render: (cell: string) => {
+          return `${cell} ms`;
+        },
+      },
+      {
+        title: 'Min',
+        dataIndex: 'min',
+        sorter: sorterFunction('min'),
+        render: (cell: string) => {
+          return `${cell} ms`;
+        },
+      },
+      {
+        title: 'Max',
+        dataIndex: 'max',
+        sorter:sorterFunction('max'),
+        render: (cell: string) => {
+          return `${cell} ms`;
+        },
+      },
+      {
+        title: 'ST Total',
+        dataIndex: 'selfTotal',
+        sorter:sorterFunction('selfTotal'),
+        render: (cell: string) => {
+          return `${cell} ms`;
+        },
+      },
+      {
+        title: 'ST Avg',
+        dataIndex: 'selfAvg',
+        sorter: sorterFunction('selfAvg'),
+        render: (cell: string) => {
+          return `${cell} ms`;
+        },
+      },
+      {
+        title: 'ST Min',
+        dataIndex: 'selfMin',
+        sorter:sorterFunction( 'selfMin' ),
+        render: (cell: string) => {
+          return `${cell} ms`;
+        },
+      },
+      {
+        title: 'ST Max',
+        dataIndex: 'selfMax',
+        sorter:sorterFunction('selfMax'),
+        render: (cell: string) => {
+          return `${cell} ms`;
+        },
+      },
+      {
+        title: 'ST in Duration',
+        dataIndex: 'percent',
+        sorter:sorterFunction('percent'),
+        render: (cell: string) => {
+          return `${cell} %`;
+        },
+      },
+    ];
+
+    columns = columns.map(val => {
+      return { ...val, onCell: record => onCellFunction(record) };
+    });
+
+    const findTablesAccToSelectors = (tableValue: ITableSpan[]): ITableSpan[] => {
```

Comment:
Can you describe in words what this function is doing? The name seems very non descriptive

Reply:
This functions makes the datasource for the antd table according to the valuenameselector values. Like it adds the child attribute for each element in datasource if we give valuenameselector2 (sub-grouping) . So this function finally outputs what are the main rows and what are the sub-group rows of  each main row

Reply:
```suggestion
    /**
    * Pre-process the table data into groups and sub-groups
    */
    const findTablesAccToSelectors = (tableValue: ITableSpan[]): ITableSpan[] => {
```